src/Control/Effect/Internal/Error.hs

```diff
@@ -234,6 +234,19 @@ errorToErrorIO main = do
     else
       throwIO exc
 
+errorToErrorIO' :: forall e m a. (Exception e, Effs '[ErrorIO, Embed IO] m) =>
+   InterpretErrorC e m a
+   -> m (Either e a)
```

Comment:
Since `e` is being thrown around as an `IO` exception, another alternative would be to not do the outermost `catchIO` , returning `a` instead of `Either e a`. That'd would demand that the user must manually do a `catchIO` on `e` if they want to be sure any exception doesn't escape.
What's your opinion on this?

Reply:
I think that is more desirable since uncaught exceptions will probably get rethrown anyway.

Reply:
I agree, but I would be ok either way. Pick the one you like best!